base/linalg/uniformscaling.jl

```diff
@@ -313,3 +313,28 @@ function hvcat(rows::Tuple{Vararg{Int}}, A::Union{AbstractVecOrMat,UniformScalin
     end
     return hvcat(rows, promote_to_arrays(n,1, promote_to_array_type(A), A...)...)
 end
+
+
+## Cholesky
+function _chol!(J::UniformScaling, uplo)
```

Comment:
It is a trick to allow a recursive definition of `chol`. See https://github.com/JuliaLang/julia/blob/71d7c168dc1412a030e1de5c3e1c98720d390b17/base/linalg/cholesky.jl#L75. The outer structures will typically be mutable but not the inner. I think it is fine because `!` is not saying "will mutate", just "be aware, might mutate".